### PR TITLE
Feature/sticky atm

### PR DESCRIPTION
**Pull request by KsoonQuea**
Update the changes of strategy place order

### CODE REVIEW

**Comment by KsoonQuea** [comment]
Open PR of  strategy place order

**Comment by KsoonQuea** [comment]
update PR

File: config.py [hunk 1]
```diff
@@ -36,4 +36,8 @@ def __init__(self) -> None:
         self.TELEGRAM_API=os.getenv('TELEGRAM_API')
         self.TELEGRAM_CHANNEL=os.getenv('TELEGRAM_CHANNEL')
         self.TELEGRAM_PNL_TOPIC=os.getenv('TELEGRAM_PNL_TOPIC')
-        self.TELEGRAM_TRADECALL_TOPIC=os.getenv('TELEGRAM_TRADECALL_TOPIC')
\ No newline at end of file
+        self.TELEGRAM_TRADECALL_TOPIC=os.getenv('TELEGRAM_TRADECALL_TOPIC')
+        
+        # Ks Added
+        self.STRATEGIES_TABLE = "strategies"
```
Comment (kingsleylow0327) [inline config.py]: make sure that all are in caps and python variable matching env variable
remove empty lines and #Ks Added also

File: handler/bingx_stratery_place_order.py [hunk 2]
```diff
@@ -27,29 +27,69 @@ def calculate_qty(wallet, entry_price, sl, percentage):
...
+    
+    return True
 
 def h_bingx_strategy_order(dbcon, order_json, player_id, message_id):
-    json_ret = {"msg": "Order Placed"}
-    json_ret["error"] = []
+    json_ret            = {"msg": "Order Placed"}
+    json_ret["error"]   = []
+    
+    try:
```
Comment (kingsleylow0327) [inline handler/bingx_stratery_place_order.py]: this chuck here seems wont hit, in actual scenario

**Review by kingsleylow0327 — COMMENTED** [review]
please take a look

File: sql_con.py [hunk 3]
```diff
@@ -107,18 +107,32 @@ def get_player_api(self, player_id):
...
+        and f.type = {}
         and
         (a.expiry_date is null
         or
         a.expiry_date > now())
-        order by role DESC""".format(self.config.API_TABLE, self.config.FOLLOWER_TABLE, self.config.GROUP_TRADER_TABLE, player_id, player_id, platform)
+        order by role DESC""".format(self.config.API_TABLE, self.config.FOLLOWER_TABLE, self.config.GROUP_TRADER_TABLE, player_id, player_id, platform, order_type)
+        return self.dbcon_manager(sql, get_all=True)
+    
+    def get_strategy_follower(self, strategy_id, platform, order_type = 1):
```
Comment (kingsleylow0327) [inline sql_con.py]: join this function

**Comment by KsoonQuea** [comment]
Resolve last meeting concerns

File: sql_con.py [hunk 4]
```diff
@@ -225,4 +240,27 @@ def get_order_by_ref_id(self, last_month, this_month):
         return self.dbcon_manager(sql, get_all=True)
 
     def close_cursor(self):
-        self.cursor.close()
\ No newline at end of file
+        self.cursor.close()
+    
+    def get_all_strategies(self):
```
Comment (kingsleylow0327) [inline sql_con.py]: remove this

File: sql_con.py [hunk 5]
```diff
@@ -225,4 +240,27 @@ def get_order_by_ref_id(self, last_month, this_month):
...
+        sql = """
+        Select id, name from {}
+        where deleted_at is null
+        """.format(
+            self.config.STRATEGY_TABLE
+        )
+        
+        return self.dbcon_manager(sql, get_all=True)
+    
+    def get_strategy_where(self, attribute, data):
```
Comment (kingsleylow0327) [inline sql_con.py]: ditto

File: app.py [hunk 6]
```diff
@@ -377,8 +377,17 @@ async def on_message(message):
                 for error in spilt_discord_message(ret.get("error")):
                     await thread.send(error)
                     logger.info(error)
-            await thread.send("Order SUCCESSFUL ✅ \n")
-            thread_message = f'🟢 {cur_date} -- {order.get("coin_pair")} {order.get("coin_pair")}'
+                    
+                if ret.get("status") and ret.get("status") == 400:
+                    await thread.send("Order Skipped ❌ \n")
```
Comment (kingsleylow0327) [inline app.py]: You forgot the detail message that i mentioned to you last time

**Review by kingsleylow0327 — COMMENTED** [review]
please take a look

File: sql_con.py (hunk 4 above)
Comment (KsoonQuea) [reply sql_con.py]: Removed and push (16 Sept)

File: sql_con.py (hunk 5 above)
Comment (KsoonQuea) [reply sql_con.py]: Removed and push (16 Sept)

File: app.py (hunk 6 above)
Comment (KsoonQuea) [reply app.py]: Add reason details at behind of message (16 Sept)

**Comment by KsoonQuea** [comment]
Update PR

**Review by kingsleylow0327 — APPROVED** [review]
LGTM

**Pull request merged**